diff --git a/recaptcha.js b/recaptcha.js
new file mode 100644
index 0000000..41645b3
--- /dev/null
+++ b/recaptcha.js
@@ -0,0 +1,28 @@
+(function ($, window, Drupal) {
+
+  Drupal.behaviors.recaptcha = {
+    attach: function (context) {
+      $('.g-recaptcha', context).each(function() {
+        if (typeof grecaptcha === 'undefined') {
+          return;
+        }
+        if ($(this).hasClass('recaptcha-processed')) {
+          grecaptcha.reset();
+        }
+        else {
+          grecaptcha.render(this, $(this).data());
+          $(this).addClass('recaptcha-processed');
+        }
+
+      });
+    }
+  };
+
+  window.drupalRecaptchaOnload = function() {
+    $('.g-recaptcha').each(function() {
+      grecaptcha.render(this, $(this).data());
+      $(this).addClass('recaptcha-processed');
+    });
+  }
+})(jQuery, window, Drupal);
+
diff --git a/recaptcha.module b/recaptcha.module
index 1e65aab..190b925 100644
--- a/recaptcha.module
+++ b/recaptcha.module
@@ -114,20 +114,16 @@ function recaptcha_captcha($op, $captcha_type = '') {
           $captcha['form']['recaptcha_widget'] = array(
             '#markup' => '<div' . drupal_attributes($attributes) . '></div>',
             '#suffix' => $noscript,
-          );
-
-          // @todo: #1664602: D7 does not yet support "async" in drupal_add_js().
-          // drupal_add_js(url('https://www.google.com/recaptcha/api.js', array('query' => array('hl' => $language->language), 'absolute' => TRUE)), array('defer' => TRUE, 'async' => TRUE, 'type' => 'external'));
-          $data = array(
-            '#tag' => 'script',
-            '#value' => '',
-            '#attributes' => array(
-              'src' => url('https://www.google.com/recaptcha/api.js', array('query' => array('hl' => $language->language), 'absolute' => TRUE)),
-              'async' => 'async',
-              'defer' => 'defer',
+            '#attached' => array(
+              'js' => array(
+                drupal_get_path('module', 'recaptcha') . '/recaptcha.js',
+                array(
+                  'data' => url('https://www.google.com/recaptcha/api.js', array('query' => array('hl' => $language->language, 'render' => 'explicit', 'onload' => 'drupalRecaptchaOnload'), 'absolute' => TRUE)),
+                  'type' => 'external',
+                ),
+              ),
             ),
           );
-          drupal_add_html_head($data, 'recaptcha_api');
         }
         else {
           // Fallback to Math captcha as reCAPTCHA is not configured.
diff --git a/recaptcha.test b/recaptcha.test
index 4afedf7..7bc7450 100644
--- a/recaptcha.test
+++ b/recaptcha.test
@@ -128,7 +128,7 @@ class ReCaptchaBasicTest extends DrupalWebTestCase {
     // Check if there is a reCAPTCHA on the login form.
     $this->drupalGet('user');
     $this->assertRaw($grecaptcha, '[testReCaptchaOnLoginForm]: reCAPTCHA is shown on form.');
-    $this->assertRaw('<script src="https://www.google.com/recaptcha/api.js?hl=' . $language->language . '" async="async" defer="defer"></script>', '[testReCaptchaOnLoginForm]: reCAPTCHA is shown on form.');
+    $this->assertRaw('<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?hl=' . $language->language . '&amp;render=explicit&amp;onload=-"></script>', '[testReCaptchaOnLoginForm]: reCAPTCHA is shown on form.');
     $this->assertNoRaw($grecaptcha . '<noscript>', '[testReCaptchaOnLoginForm]: NoScript code is not enabled for the reCAPTCHA.');
 
     // Test if the fall back url is properly build and noscript code added.
